pkg/controller/capacity/capacity_controller.go

```diff
@@ -41,64 +39,83 @@ const (
 	AgentName   = "capacity-controller"
 	SadPodLimit = 5
 
-	ClustersNotReady = "ClustersNotReady"
-	InProgress       = "InProgress"
-	InternalError    = "InternalError"
-	PodsNotReady     = "PodsNotReady"
-	DeploymentStuck  = "DeploymentStuck"
+	InProgress      = "InProgress"
+	InternalError   = "InternalError"
+	PodsNotReady    = "PodsNotReady"
+	DeploymentStuck = "DeploymentStuck"
 
-	CapacityTargetConditionChanged  = "CapacityTargetConditionChanged"
-	ClusterCapacityConditionChanged = "ClusterCapacityConditionChanged"
+	CapacityTargetConditionChanged = "CapacityTargetConditionChanged"
 )
 
 // Controller is the controller implementation for CapacityTarget resources
 type Controller struct {
-	clientset shipperclient.Interface
-	store     clusterclientstore.Interface
+	shipperClient shipperclient.Interface
+	kubeClient    kubernetes.Interface
 
 	capacityTargetsLister listers.CapacityTargetLister
 	capacityTargetsSynced cache.InformerSynced
 
-	releasesLister       listers.ReleaseLister
-	releasesListerSynced cache.InformerSynced
+	deploymentsLister appslisters.DeploymentLister
+	deploymentsSynced cache.InformerSynced
+
+	podsLister corelisters.PodLister
+	podsSynced cache.InformerSynced
 
 	workqueue workqueue.RateLimitingInterface
-	recorder  record.EventRecorder
+
+	recorder record.EventRecorder
 }
 
 // NewController returns a new CapacityTarget controller.
 func NewController(
-	clientset shipperclient.Interface,
+	kubeClient kubernetes.Interface,
+	kubeInformerFactory kubeinformers.SharedInformerFactory,
+	shipperClient shipperclient.Interface,
```

Comment:
Would an object conforming to `ClientsetInterface` be helpful here?

Reply:
As matter of fact, it would. Thanks for the suggestion! I'll apply it together with the upcoming changes to the traffic controller.